packages/teleport-project-generator/src/file-handlers.ts

```diff
@@ -240,26 +243,31 @@ const createHTMLEntryFileChunks = (uidl: ProjectUIDL, options: EntryFileOptions)
     }
 
     // inline style
-    if (asset.type === 'style' && asset.content) {
+    if (asset.type === 'style' && 'content' in asset) {
       const styleTag = HASTBuilders.createHTMLNode('style')
       HASTUtils.addTextNode(styleTag, asset.content)
       HASTUtils.addChildNode(headNode, styleTag)
     }
 
     // script (external or inline)
     if (asset.type === 'script') {
-      const scriptInBody = (asset.options && asset.options.target === 'body') || false
+      let scriptInBody = false
+      if ('options' in asset) {
+        if ('target' in asset.options && asset.options.target === 'body') {
+          scriptInBody = true
+        }
+      }
       const scriptTag = HASTBuilders.createHTMLNode('script')
       HASTUtils.addAttributeToNode(scriptTag, 'type', 'text/javascript')
       if (assetPath) {
         HASTUtils.addAttributeToNode(scriptTag, 'src', assetPath)
-        if (asset.options && asset.options.defer) {
+        if ('options' in asset && asset.options.defer) {
           HASTUtils.addBooleanAttributeToNode(scriptTag, 'defer')
         }
-        if (asset.options && asset.options.async) {
+        if ('options' in asset && asset.options.async) {
           HASTUtils.addBooleanAttributeToNode(scriptTag, 'async')
         }
-      } else if (asset.content) {
+      } else if ('content' in asset) {
```

Comment:
same syntax as above

Reply:
`asset?.content` throws typescript error that `Property 'content' does not exist on type 'UIDLScriptAsset'.
  Property 'content' does not exist on type 'UIDLScriptExternalAsset'`

Reply:
It should be something like this 

```js
if (asset.type === 'script') {
      const scriptInBody = (asset.options && asset.options.target === 'body') || false
      const scriptTag = HASTBuilders.createHTMLNode('script')
      HASTUtils.addAttributeToNode(scriptTag, 'type', 'text/javascript')

      if (assetPath && asset.path) {
        HASTUtils.addAttributeToNode(scriptTag, 'src', assetPath)
      }

      if (assetPath && asset.content) {
        HASTUtils.addTextNode(scriptTag, asset.content)
      }

      if (asset.options && asset.options.defer) {
        HASTUtils.addBooleanAttributeToNode(scriptTag, 'defer')
      }
      if (asset.options && asset.options.async) {
        HASTUtils.addBooleanAttributeToNode(scriptTag, 'async')
      }

      if (scriptInBody) {
        HASTUtils.addChildNode(bodyNode, scriptTag)
      } else {
        HASTUtils.addChildNode(headNode, scriptTag)
      }
    }
```

And by the way, `options` should be present for `UIDLScriptInlineAsset`. Because, any script tag can have options `async` | `defer` etc. But as the same as styles, it can't have both `src` of remote and script content at the same time. It's just a `syntatical` difference but mostly they have the same options and everything 😄